src/libcore/intrinsics.rs

```diff
@@ -1401,4 +1401,22 @@ extern "rust-intrinsic" {
     /// Emits a `!nontemporal` store according to LLVM (see their docs).
     /// Probably will never become stable.
     pub fn nontemporal_store<T>(ptr: *mut T, val: T);
+
+    /// Initialize the arglist `ap` for processing via `va_arg`.
+    #[cfg(not(stage0))]
+    pub fn va_start(ap: *mut i8);
+
+    /// Destroy the arglist `ap` after initialization with `va_start` or
+    /// `va_copy`.
+    #[cfg(not(stage0))]
+    pub fn va_end(ap: *mut i8);
+
+    /// Copy the current location of arglist `ap0` to the arglist `ap1`.
+    #[cfg(not(stage0))]
+    pub fn va_copy(dst: *mut i8, src: *const i8);
+
+    /// Loads an argument of type `T` from the `va_list` `ap` and increment the
+    /// argument `ap` points to.
+    #[cfg(not(stage0))]
+    pub fn va_arg<T: Copy>(ap: *mut i8) -> T;
```

Comment:
These intrinsics could take the `VaList` type, which could be made `#[repr(transparent)]`, I think, just to increase their safety a bit.

Reply:
The `va_start`, `va_end`, and `va_copy` intrinsics explicitly take an `i8*`. https://llvm.org/docs/LangRef.html#llvm-va-start-intrinsic also see the example https://llvm.org/docs/LangRef.html#variable-argument-handling-intrinsics

Reply:
That can be handled by casting `rustc_trans` - assuming they are even required to have `i8*` type in the actual LLVM implementation.

Reply:
Cool, I'll give it a shot.